### PR TITLE
Add `InvalidPayload` status

### PR DESCRIPTION
**Pull request by kaapstorm**
## Technical Summary

Adds an `InvalidPayload` state to `RepeatRecord` and `RepeatRecordAttempt` in order to make better decisions about invalid payloads.

"Invalid Payload" in the Repeat Records Report:

image

**NOTE:** The base branch is `gh/repeaters/refactor-error-handling` until PR https://github.com/dimagi/commcare-hq/pull/34998 is merged.

### Safety story

Tested locally.

### Automated test coverage

Includes test coverage.

### Migrations

- [x] The migrations in this code can be safely applied first independently of the code

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/motech/repeaters/views/tests/test_repeat_records.py
Comment (millerdev) [inline corehq/motech/repeaters/views/tests/test_repeat_records.py:118]: re: comment above. Do template(s) that reference `count_State` need to be updated?

https://github.com/dimagi/commcare-hq/blob/acf5951647819fe41ce6b1e514ab5137dd9a663d/corehq/motech/repeaters/templates/repeaters/partials/repeater_row.html#L11-L26

**Review by millerdev — DISMISSED** [review]

File: corehq/motech/repeaters/views/tests/test_repeat_records.py
Comment (kaapstorm) [reply corehq/motech/repeaters/views/tests/test_repeat_records.py:118]: At first I added a new column for invalid payloads, but the table didn't look good for repeaters with long names. Considering we already exclude the Empty state, I thought it would be better to exclude the Invalid Payload state too.

But on second thoughts that seems wrong. So I've included the Invalid Payload state with Cancelled (and Empty with Success). https://github.com/dimagi/commcare-hq/pull/34999/commits/d6924defebd4313389d5e21c401a1d0526a3f52b

There is a downside though. The link for the count of "InvalidOrCancelled" opens the repeat records report with the filter set to Cancelled. It's not going to be obvious to users that they need to change the filter to "Invalid Payloads" to see the other repeat records. What do you think? Another column? Or Success-ish and Cancelled-ish?

File: corehq/motech/repeaters/models.py
Comment (gherceg) [inline corehq/motech/repeaters/models.py:506]: So this is basically saying if this is something like a 401 or 404, it is a payload error and we should just cancel that record immediately right?

File: corehq/motech/repeaters/models.py
Comment (gherceg) [inline corehq/motech/repeaters/models.py:1238]: Until I saw it used, I had no idea what `tb_str` was 😅. Not sure if that is an indication that it should be a more explicit name or not. I'll leave that up to you.

File: corehq/motech/repeaters/views/tests/test_repeat_records.py
Comment (gherceg) [reply corehq/motech/repeaters/views/tests/test_repeat_records.py:118]: I don't have much to add here, but I think this table is already notoriously confusing for users. I don't mind decoupling the states that a user sees from the states that exist in the backend, which I think would be success-ish and cancelled-ish in this case.

File: corehq/motech/repeaters/models.py
Comment (kaapstorm) [reply corehq/motech/repeaters/models.py:1238]: Oh, that is an excellent indication that it should be a more explicit name. :grin: Thank you.

File: corehq/motech/repeaters/models.py
Comment (kaapstorm) [reply corehq/motech/repeaters/models.py:506]: Yes exactly. We retry if there is a 5XX error or a 4XX error that might succeed if we try again later, but if the problem is the payload, retrying isn't going to help.

File: corehq/motech/repeaters/models.py
Comment (kaapstorm) [reply corehq/motech/repeaters/models.py]: https://github.com/dimagi/commcare-hq/pull/34999/commits/42192777ca7d6593d38748ebdb390473b271203e

File: corehq/motech/repeaters/models.py
Comment (kaapstorm) [inline corehq/motech/repeaters/models.py:68]: **Today I Learned** that if a test passes locally but fails on GitHub, it could be because, when GitHub merges in master before it runs the tests, that merge deletes an import that you need!

**Review by millerdev — APPROVED** [review]

**Pull request merged**
